kaldb/src/test/java/com/slack/kaldb/metadata/service/ServiceMetadataSerializerTest.java

```diff
@@ -0,0 +1,131 @@
+    final Instant partitionStart = Instant.now();
+    final Instant partitionEnd = Instant.now().plus(1, ChronoUnit.DAYS);
+    final String partitionName = "partitionName";
+    final List<String> partitionList = List.of(partitionName);
+
+    final String name = "testService";
+    final String owner = "testOwner";
+    final long throughput = 2000;
+    final List<ServicePartitionMetadata> list =
+        Collections.singletonList(
+            new ServicePartitionMetadata(
+                partitionStart.toEpochMilli(), partitionEnd.toEpochMilli(), partitionList));
+    final ServiceMetadata serviceMetadata = new ServiceMetadata(name, owner, throughput, list);
+
+    String serializedServiceMetadata = serDe.toJsonStr(serviceMetadata);
+    assertThat(serializedServiceMetadata).isNotEmpty();
+
+    ServiceMetadata deserializedServiceMetadata = serDe.fromJsonStr(serializedServiceMetadata);
+    assertThat(deserializedServiceMetadata).isEqualTo(serviceMetadata);
+
+    assertThat(deserializedServiceMetadata.name).isEqualTo(name);
+    assertThat(deserializedServiceMetadata.partitionConfigs).isEqualTo(list);
+  }
+
+  @Test
+  public void testServiceMetadataSerializerDuplicatePartitions()
+      throws InvalidProtocolBufferException {
+    final Instant partitionStart1 = Instant.now();
+    final Instant partitionEnd1 = Instant.now().plus(1, ChronoUnit.DAYS);
+    final Instant partitionStart2 = partitionEnd1.plus(1, ChronoUnit.MILLIS);
+    final Instant partitionEnd2 = partitionStart2.plus(1, ChronoUnit.DAYS);
+    final String partitionName = "partitionName1";
+    final List<String> partitionList = List.of(partitionName);
+
+    final String name = "testService";
+    final String owner = "testOwner";
+    final long throughput = 2000;
+    final List<ServicePartitionMetadata> list =
+        List.of(
+            new ServicePartitionMetadata(
+                partitionStart1.toEpochMilli(), partitionEnd1.toEpochMilli(), partitionList),
+            new ServicePartitionMetadata(
+                partitionStart2.toEpochMilli(), partitionEnd2.toEpochMilli(), partitionList));
+    final ServiceMetadata serviceMetadata = new ServiceMetadata(name, owner, throughput, list);
+
+    String serializedServiceMetadata = serDe.toJsonStr(serviceMetadata);
+    assertThat(serializedServiceMetadata).isNotEmpty();
+
+    ServiceMetadata deserializedServiceMetadata = serDe.fromJsonStr(serializedServiceMetadata);
+    assertThat(deserializedServiceMetadata).isEqualTo(serviceMetadata);
+
+    assertThat(deserializedServiceMetadata.name).isEqualTo(name);
+    assertThat(deserializedServiceMetadata.partitionConfigs).isEqualTo(list);
+  }
+
+  @Test
+  public void testServiceMetadataSerializerEmptyList() throws InvalidProtocolBufferException {
+    final String name = "testService";
+    final String owner = "testOwner";
+    final long throughput = 2000;
+    final List<ServicePartitionMetadata> list = Collections.emptyList();
```

Comment:
Why not throw an exception on an empty partition list?

Reply:
I initially did have checks for an empty partition list, but removed them for now. I believe that we will have a few scenarios where we will want the ability define a service without immediately defining a partition map.